### PR TITLE
Dynamic space world generation and debris.

### PR DESCRIPTION
**Pull request by moonheart08**
## About the PR
This impacts exactly nothing, as it's simply not enabled on Wizard's Den servers at the minute. Requested by @metalgearsloth 

**Media**
image
image

### CODE REVIEW

**Comment by moonheart08** [comment]
DNM until electro looks at the gamestate crash.

**Comment by EmoGarbage404** [comment]
Why was this closed?

**Comment by mirrorcult** [comment]
image

lol

**Comment by Arteben** [comment]
Are player movements limited? At large coordinates in deep space, everything works not very correctly, for example, at 100000000 100000000 (you can use tp command)

**Comment by EmoGarbage404** [comment]
> Are player movements limited? At large coordinates in deep space, everything works not very correctly, for example, at 100000000 100000000 (you can use tp command)

No but thats not really relevant to the pr at hand.

**Comment by moonheart08** [comment]
> Are player movements limited? At large coordinates in deep space, everything works not very correctly, for example, at 100000000 100000000 (you can use tp command)

Out of scope as stated.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

File: Content.IntegrationTests/Tests/EntityTest.cs [hunk 1]
```diff
@@ -161,6 +164,10 @@ public async Task AllComponentsOneToOneDeleteTest()
                 "MapGrid",
                 "StationData", // errors when removed mid-round
                 "Actor", // We aren't testing actor components, those need their player session set.
+                "BlobFloorPlanBuilder", // Implodes if unconfigured.
+                "DebrisFeaturePlacerController", // Above.
+                "LoadedChunk", // Worldgen chunk loading malding.
+                "BiomeSelection", // Whaddya know, requires config.
```
[the remark this answers is not in the document]
Comment (moonheart08) [reply Content.IntegrationTests/Tests/EntityTest.cs:170]: Not sure how to address this.

File: Content.Server/Worldgen/GridPointsNearEnumerator.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using System.Diagnostics.Contracts;
+
+namespace Content.Server.Worldgen;
+
+/// <summary>
+///     A struct enumerator of points on a grid within the given radius.
+/// </summary>
+public struct GridPointsNearEnumerator
+{
```
[the remark this answers is not in the document]
Comment (moonheart08) [reply Content.Server/Worldgen/GridPointsNearEnumerator.cs:10]: I try not to immediately move APIs to engine as that locks them in stone.

File: Content.Server/Worldgen/Tools/PoissonDiskSampler.cs [hunk 3]
```diff
@@ -0,0 +1,194 @@
+﻿using Robust.Shared.Random;
+
+namespace Content.Server.Worldgen.Tools;
+
+/// <summary>
```
Comment (metalgearsloth) [inline Content.Server/Worldgen/Tools/PoissonDiskSampler.cs:5]: I keep getting arithmetic overflows when I try to use this though not sure where my inputs were going wrong.

Comment (moonheart08) [reply Content.Server/Worldgen/Tools/PoissonDiskSampler.cs:5]: it's a bit annoying but tl;dr the input bounds must be in a local coordinate space relative to 0,0.

**Pull request merged**